android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -2444,6 +2447,38 @@ public void showIntroductionScreenForDeeplink(@NonNull String deepLink,
     IntroductionDialogFragment.show(getSupportFragmentManager(), deepLink, factory);
   }
 
+  @Override
+  public boolean onKeyUp(int keyCode, KeyEvent event)
+  {
+    switch (keyCode)
+    {
+      case KeyEvent.KEYCODE_DPAD_DOWN:
+        Statistics.INSTANCE.trackEvent(Statistics.EventName.ZOOM_OUT);
+        MapFragment.nativeScaleMinus();
+        return true;
+      case KeyEvent.KEYCODE_DPAD_UP:
```

Comment:
Why we need to process this event types?

Reply:
This event is fired when hardware button up/down/escape is pressed. Please take a look at demo https://www.youtube.com/watch?v=2QHUiheVF5c&feature=youtu.be .

Reply:
I meant why we need to process zoom in/out events? Why we need it now?

Reply:
And how it worked before

Reply:
It didn't work before, because we didn't have a need for it. Now, we need to be able to intercept such events at least to make motobike controls work.

Reply:
oh, I understood, thank you.